base_layer/key_manager/src/cipher_seed.rs

```diff
@@ -230,25 +243,29 @@ impl CipherSeed {
     }
 
     fn apply_stream_cipher(data: &mut Vec<u8>, passphrase: &str, salt: &[u8]) -> Result<(), KeyManagerError> {
-        let argon2 = Argon2::default();
         let blake2_nonce_hasher: VarBlake2b = VarBlake2b::new(size_of::<Nonce>())
             .expect("Should be able to create blake2 hasher; will only panic if output size is 0 or greater than 64");
 
         let mut encryption_nonce = [0u8; size_of::<Nonce>()];
+
+        // TODO: use hashing API to generate suitable domain separation instead
+        let label = b"chacha20_salt";
         blake2_nonce_hasher
+            .chain(label)
```

Comment:
A safer approach (and one used by the hashing API) is to ensure that any variable-length input to the underlying hash function be prepended with a fixed-length encoding of the input's length. This avoids accidental or malicious collisions. If you're taking a belt-and-suspenders approach, it's best to do this even for fixed-length inputs, in case they change later.

Reply:
I refactored the code to use the hashing API and take the first `n` byte elements, depending on the use case.